src/Smart.Design.Console/CommandLineParser.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using CommandLine;
+using FluentValidation;
+using Smart.Design.Console.CommandOptions;
+using Smart.Design.Console.Models;
+using Smart.Design.Console.Models.Validations;
+using Smart.Design.Console.UseCases;
+
+namespace Smart.Design.Console
+{
+    public class CommandLineParser
+    {
+        private readonly string[] _args;
+
+        public CommandLineParser(string[] _args)
+        {
+            this._args = _args;
+        }
+
+        public CommandResponse ExecuteCommandWithArguments()
```

Comment:
- I was thinking that having a method called "ExecuteCommand" is a bit weid on a class that is called a "Parser", but I don't really have a better name.
- I'm wondering if adding "WithArguments" or even "CommandWithArguments" adds any value to the name.

Reply:
Hmm I see your point. I chose ExecuteCommand because it's really executing a "command" in the CLI sense of the term (so not a "command" as any generic name like "Run" or "Start"). If I removed the "WithArguments" I would need to add it in the signature of the method and remove it from the constructor because that could be confusing.

Reply:
I will think about it